### ISSUE TITLE
Bug Report: Rmd file A does not run when Rmd file B has wrongly specified YAML header

### ISSUE THREAD

**Issue by S-UP** [rstudio/rmarkdown#1005]
Hi,

I've identified a rather strange bug.

Below code cannot be executed when there is an Rmd file present in the very same folder, but with a wrongly specified YAML header (e.g. white space before 'title' in the YAML section).

**File A**

```  
  ---
  title: "Something Is Wrong"
  output: html_document
  runtime: shiny
  ---
  
  Hello World
  
```

**File B**

```  
  ---
      title: "Something Is Wrong"
  output: html_document
  runtime: shiny
  ---
  
  Hello World
  
```

This gives me an error message when running **File A**:

> Error in yaml::yaml.load(string, ...) : 
>   Parser error: did not find expected <document start> at line 2, column 1
> Calls: <Anonymous> ... parse_yaml_front_matter -> yaml_load_utf8 -> <Anonymous> -> .Call
> Execution halted

**Comment by jjallaire** [comment]
There may be an _output.yml or _site.yml file in your directory which has
some invalid YAML

**Comment by yihui** [comment]
I cannot reproduce this issue by putting these two files under an empty directory, so please provide a full reproducible example (you can upload a zip file here on Github).

**Comment by gemish** [comment]
Same issue here, I think.  I have two Rmd files in an otherwise empty folder.  There's only a header in each:

Test

```
---
title: "test1"
date: "December 18, 2017"
output: html_document
  runtime: shiny
---
```

Test 2
```
---
title: "test2"
date: "December 18, 2017"
output: html_document
runtime: shiny  
---

```
If I try to run Test 2 while the invalid spacing is in Test, I get this error:

```
Error in yaml::yaml.load(string, ...) : 
  Scanner error: mapping values are not allowed in this context at line 4, column 10
Calls: <Anonymous> ... parse_yaml_front_matter -> yaml_load_utf8 -> <Anonymous> -> .Call
Execution halted
```


If I remove the space from Test, save, and re-run Test 2, it runs fine.  Zipped files attached.

[testforymlerror.zip](https://github.com/rstudio/rmarkdown/files/1569910/testforymlerror.zip)

**Comment by yihui** [comment]
Thanks @gemish for the excellent reproducible example. I was able to reproduce the issue, but I'd say it was be design. Please see the Details section of the help page `?rmarkdown::run`. In your case, all Rmd files have to be scanned to determine the `default_file`.

The solution is to correct the YAML metadata of your test.Rmd (remove the white spaces before `runtime: shiny`). In @S-UP's case, remove the white spaces before `title`.

**Comment by gorodnichy** [comment]
I had exactly the same problem.It looked really steange
In two different directories I did exactly the same to create a new .Rmd with shiny file from template:
File->New File ->R Markdown... ----> X Shiny.

Then in one directory it would run ok. In the other it produced this error:
>   Error in yaml::yaml.load(string, ...) : 
>   Scanner error: mapping values are not allowed in this context at line 4, column 10
> Calls: <Anonymous> ... parse_yaml_front_matter -> yaml_load_utf8 -> <Anonymous> -> .Call
> Execution halted

Then I found this post and the THIS EXPLANATION:  _In your case, all Rmd files have to be scanned to determine the default_file._

So I moved all my other Rmd files away from this directory, and then it worked fine!
